### DISCUSSION TITLE
Running flux commands from slack

### DISCUSSION THREAD

**Discussion by havard024** [fluxcd/flux2#3561] [Proposals] [answered]
We've been using https://botkube.io/ to move most of our devops operations to slack and we love it.

Now that we've moved over to use flux for our deployments, we were wondering if there have been any thoughts / discussions on creating a slack app to expose flux commands from slack or to integrate with existing tools like botkube.

**Comment by stefanprodan** [comment, marked as answer]
Flux integrates with Slack for [events and alerts](https://fluxcd.io/flux/components/notification/alert/). I'm not sure how useful would be to call Flux commands from Slack, it goes against what we're trying to do with the Flux project. If you change the cluster state using a bot, then this will conflict with Flux and it will undo all changes. Also having a Slack bot that can connect to your production systems seems like a major security vulnerability to me, in a GitOps world, the Kubernetes API is not exposed to the outside world, operators have only read-only access and all changes go via Git using PRs.

**Reply by havard024** [reply]
Our use case is basically to be able to run `flux reconcile` from slack, in case the automated pipeline we have setup fails. With a slack integration, we are not dependent on our laptop to trigger the flux pipeline.

I agree that changing the cluster through slack or any other means than git is a very bad idea, we are only interested in read commands to be able to check the status or trigger the flux pipeline.

We only use botkube to check the status of the cluster, have joint debug sessions and another benefit is we can share knowledge of debugging through looking at how others interact with kubernetes through the botkube. 

Security wise, botkube does not expose the kubernetes API directly to the outside world, there is an application running in the cluster that is responsible for listening to slack commands and based on the command, use the internal kubernetes API and send the response back to slack.

Botkube currently supports kubectl and helm, but it is possible to add plugins for other commands, like flux.

**Reply by stefanprodan** [reply]
Ok cool, besides `flux reconcile` you could also call `flux suspend/resume` commands which are helpful during an incident or if you just want to freeze deployments over the weekend.

**Reply by havard024** [reply]
That's perfect, thanks for the discussion, this feature probably fits better as a plugin to botkube at the moment, I'll try to make the plugin myself 👍

**Reply by makkes** [reply]
If you'd like a more extensive UI experience, you can also take a look at [Weave GitOps](https://github.com/weaveworks/weave-gitops/) which supports both, manually kicking off a reconciliation loop as well as suspending/resuming resources.
